operator/src/test/java/org/keycloak/operator/testsuite/utils/K8sUtils.java

```diff
@@ -113,37 +133,37 @@ public static String inClusterCurl(KubernetesClient k8sclient, String namespace,
     }
 
     public static String inClusterCurl(KubernetesClient k8sclient, String namespace, String... args) {
-        var podName = KubernetesResourceUtil.sanitizeName("curl-" + UUID.randomUUID());
+        var podName = "curl-pod";
         try {
-            Pod curlPod = k8sclient.run().inNamespace(namespace)
-                    .withNewRunConfig()
-                            .withArgs(args)
-                            .withName(podName)
-                            .withImage("curlimages/curl:7.78.0")
-                            .withRestartPolicy("Never")
-                            .done();
-            Log.info("Waiting for curl Pod to finish running");
-            Awaitility.await().atMost(3, TimeUnit.MINUTES)
-                    .until(() -> {
-                        String phase =
-                                k8sclient.pods().inNamespace(namespace).withName(podName).get()
-                                        .getStatus().getPhase();
-                        return phase.equals("Succeeded") || phase.equals("Failed");
-                    });
-
-            String curlOutput =
-                    k8sclient.pods().inNamespace(namespace)
-                            .withName(curlPod.getMetadata().getName()).getLog();
-
-            return curlOutput;
-        } catch (KubernetesClientException ex) {
-            throw new AssertionError(ex);
-        } finally {
-            Log.info("Deleting curl Pod");
-            k8sclient.pods().inNamespace(namespace).withName(podName).delete();
-            Awaitility.await().atMost(2, TimeUnit.MINUTES)
-                    .until(() -> k8sclient.pods().inNamespace(namespace).withName(podName)
-                            .get() == null);
+            Pod curlPod = new PodBuilder().withNewMetadata().withName(podName).endMetadata().withNewSpec()
+                    .addNewContainer()
+                    .withImage("curlimages/curl:8.1.2")
+                    .withCommand("sh")
+                    .withName("curl")
+                    .withStdin()
+                    .endContainer()
+                    .endSpec()
+                    .build();
+
+            try {
+                k8sclient.resource(curlPod).create();
```

Comment:
Should we rather use SSA update? What if more tests in the namespace need to use curl?

Reply:
The pod is mostly going to be immutable.  We'd have to switch to a Deployment to effectively use ssa - however it's not expected that each test will want a different version of the pod.  There's just a single definition, which only one test needs to start.  All of the others can fail with a conflict and just use the existing pod.  The only downside of this approach is that we're relying on namespace deletion to get rid of the pod.

Reply:
Can we really assume all tests in a test class (sharing the same namespace) use the same curl command? Even if now it might be true, I don't think we can make assumptions like that.

Reply:
They don't share the same command, they share the same pod.  The command is executed against the pod an exec().  We actually could get away with having just a single curl pod for all namespaces.

Reply:
Sorry, yes, you are right. The curl command is no longer the startup command.